File: src/main/java/org/jukeboxmc/item/behavior/ItemCoralFan.java
package org.jukeboxmc.item.behavior;

import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.block.Block;
import org.jukeboxmc.block.BlockType;
import org.jukeboxmc.block.behavior.BlockCoralFan;
import org.jukeboxmc.block.data.CoralColor;
import org.jukeboxmc.item.Item;
import org.jukeboxmc.item.ItemType;
import org.jukeboxmc.util.BlockPalette;
import org.jukeboxmc.util.Identifier;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class ItemCoralFan extends Item {

    private final @NotNull BlockCoralFan block;

    public ItemCoralFan( Identifier identifier ) {
        super( identifier );
        this.block = Block.create( BlockType.CORAL_FAN );
        this.blockRuntimeId = this.block.getRuntimeId();
    }

    public ItemCoralFan( ItemType itemType ) {
        super( itemType );
        this.block = Block.create( BlockType.CORAL_FAN );
        this.blockRuntimeId = this.block.getRuntimeId();
    }

    @Override
    public @NotNull ItemCoralFan setBlockRuntimeId(int blockRuntimeId ) {
        this.blockRuntimeId = blockRuntimeId;
        this.block.setBlockStates( BlockPalette.getBlockNbt( blockRuntimeId ).getCompound( "states" ) );
        return this;
    }

    public @NotNull ItemCoralFan setCoralColor(@NotNull CoralColor coralColor ) {
        this.blockRuntimeId = this.block.setCoralColor( coralColor ).getRuntimeId();
        return this;
    }

    public CoralColor getCoralColor() {
        return this.block.getCoralColor();
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/item/behavior/ItemCoralFan.kt
package org.jukeboxmc.server.item.behavior

import org.jukeboxmc.api.block.Block
import org.jukeboxmc.api.block.BlockType
import org.jukeboxmc.api.block.data.CoralColor
import org.jukeboxmc.api.item.CoralFan
import org.jukeboxmc.api.item.ItemType
import org.jukeboxmc.server.block.behavior.BlockCoralFan
import org.jukeboxmc.server.item.JukeboxItem
import org.jukeboxmc.server.util.BlockPalette

class ItemCoralFan(itemType: ItemType, countNetworkId: Boolean) : JukeboxItem(itemType, countNetworkId), CoralFan {

    private val block: BlockCoralFan = Block.create<BlockCoralFan>(BlockType.CORAL_FAN)

    override fun getColor(): CoralColor {
        return this.block.getCoralColor()
    }

    override fun setColor(coralColor: CoralColor) {
        val blockNetworkId = this.block.setCoralColor(coralColor).getNetworkId()
        this.setBlockNetworkId(blockNetworkId)
        this.block.setBlockStates(BlockPalette.getBlockNbt(blockNetworkId).getCompound("states"))
    }


}